in_toto/verifylib.py

```diff
@@ -635,6 +640,113 @@ def verify_all_item_rules(items, links, target_links=None):
         link.products, target_links)
 
 
+def verify_threshold_equality(layout, chain_link_dict):
+  """
+  <Purpose>
+    Iteratively verifies threshold conditions (if a step has been
+    performed and signed by enough number of functionaries) and link
+    artifacts (materials and products of a step is same for different
+    fucntionaries) of passed dictionary (chain_link_dict).
+
+  <Arguments>
+    layout:
+            The layout specified by the project owner against which the
+            threshold will be verified.
+
+    chain_link_dict:
+            A dictionary of key-link pair with step names as keys. For each
+            step name, there are one or more keyids and corresponding
+            link objects.
+
+  <Exceptions>
+    raises an Exception if threshold is not verified.
+    ThresholdVerificationError if the step is not performed by enough
+    functionaries or if the materials and products for a step are not same
+    for all functionaries.
+
+  <Side Effects>
+    None.
+
+  """
+
+  # We are only interested in links that are related to steps defined in the
+  # Layout, so iterate over layout.steps
+  for step in layout.steps:
+    # Skip steps that don't require multiple functionaries
+    if step.threshold <= 1:
+      continue
+
+    # Extract the key_link_dict for this step from the passed chain_link_dict
+    key_link_dict = chain_link_dict[step.name]
+
+    # take one exemplary link (e.g. the first in the step_link_dict)
+    compare_link = key_link_dict.values()[0]
```

Comment:
I don't fully understand your question.

What I was saying...
In your implementation you would get an index out of range error in case the `chain_link_dict` is empty. 

If you just move what you do in L683 a couple of lines below, you'd get a little bit of defensive programming for free, i.e. you'd probably raise your threshold verification before getting an index out of range error (independently of what we did earlier, e.g. in `verify_all_steps_signatures`).